### PR TITLE
feat: support feature flag

### LINKED ISSUE #663: Type 'Component' is not assignable to type 'ReactiveElement'. Types have separate declarations of a private property '__instanceProperties'.
Refs: https://github.com/lit/lit/issues/3241

### PR DESCRIPTION
**Pull request by himself65**
Fixes: https://github.com/toeverything/blocksuite/issues/663

### CODE REVIEW

**Review by doodlewind — APPROVED** [review]

File: packages/blocks/src/components/drag-handle.ts [hunk 1]
```diff
@@ -278,7 +281,7 @@ export class DragHandle extends LitElement {
   };
 
   override render() {
-    if (IS_PROD) return null;
+    if (getCurrentVersion() <= Version.AbbeyWood) return null;
```
Comment (doodlewind) [inline packages/blocks/src/components/drag-handle.ts:284]: `<= Version.Downhills`?

Comment (himself65) [reply packages/blocks/src/components/drag-handle.ts]: ?

Comment (himself65) [reply packages/blocks/src/components/drag-handle.ts]: Not hide in the current affine?

File: packages/blocks/src/index.ts [hunk 2]
```diff
@@ -28,23 +35,16 @@ export * from './__internal__/utils/query.js';
...
-const importIdentifier = '__ $BLOCKSUITE_BLOCKS$ __';
-
-if (env[importIdentifier] === true) {
+if (env[importIdentifier] !== undefined) {
   // https://github.com/yjs/yjs/issues/438
   console.error(
     '@blocksuite/blocks was already imported. This breaks constructor checks and will lead to issues!'
   );
+} else {
+  const isAbbeyWood = location.href.includes('pathfinder');
```
[the remark this answers is not in the document]
Comment (himself65) [reply packages/blocks/src/index.ts]: 1. Let affine set version by themselves 
2. We automatically set version in some magic ways

**Review by doodlewind — COMMENTED** [review]
I rethought and found that there may be another configuration-based approach:

``` js
flags.dragHandle = {
  'playground': true, // BlockSuite playground
  'stage': false, // pathfinder.affine.systems
  'prod': false // pathfinder.affine.pro
}
```

Since the version codenames are just used for AFFiNE marketing and can change frequently, but we always have these stable environments, just like different channels (Dev, Beta, Canary) of Chrome.

**Comment by himself65** [comment]
> I rethought and found that there may be another configuration-based approach:
> 
> ```js
> flags.dragHandle = {
>   'playground': true, // BlockSuite playground
>   'stage': false, // pathfinder.affine.systems
>   'prod': false // pathfinder.affine.pro
> }
> ```
> 
> Since the version codenames are just used for AFFiNE marketing and are fast to change, but we always have these stable environments, just like different channels (Dev, Beta, Canary) of Chrome.

Study from maskbook https://github.com/DimensionDev/Maskbook/blob/eeee66858c0736296326628dc89e0b2e97f3c3c5/packages/scripts/src/extension/flags.ts#L8

**Comment by himself65** [comment]
https://github.com/DimensionDev/Maskbook/blob/eeee66858c0736296326628dc89e0b2e97f3c3c5/packages/mask/shared/flags.ts#L10

**Comment by Himself65** [comment]
```
window.dispatchEvent(new CustomEvent('affine.set-flags', { detail: { flag: 'enable_drag_handle', value: false  }}))
```

**Review by himself65 — COMMENTED** [review]
I'm going to put this part into `store` package

**Comment by himself65** [comment]
```js
workspace.meta.awareness.setFlag('enable_drag_handle', true)
```

**Comment by himself65** [comment]
```js
workspace.meta.awareness.setFlag('enable_drag_handle', false)
```

**Pull request merged**